src/controllers/userProfileController.js

```diff
@@ -932,6 +932,24 @@ const userProfileController = function (UserProfile) {
       .catch((error) => res.status(500).send(error));
   };
 
+  const updateAllMembersTeamCode = async (req, res) => {
+    const canEditTeamCode =
+      req.body.requestor.role === 'Owner' ||
+      req.body.requestor.role === 'Administrator' ||
+      req.body.requestor.permissions?.frontPermissions.includes('editTeamCode');
```

Comment:
```suggestion
      await hasPermission(req.body.requestor, 'editTeamCode');
```
Use the `hasPermission` function to check for both role and user permissions. Also never use hardcoded role permissions (owner/admin here) so that other roles can be assigned this permission.

Reply:
okay cool! I will update it accordingly I was just following the previous way here !

Reply:
Hey! thanks for code review and updated the suggested changes!